pkg/channel/distributed/dispatcher/controller/kafkachannel.go

```diff
@@ -106,10 +111,44 @@ func NewController(
 			w.Stop()
 		}
 	}()
-
+	if err := reconciler.processManagerEvents(managerEvents); err != nil {
+		logger.Warn("Could not begin processing events from the Consumer Group Manager", zap.Error(err))
+	}
 	return reconciler.impl
 }
 
+// processManagerEvents will listen on the channel provided by the KafkaConsumerGroupManager for events
+// related to changes in the status of a managed ConsumerGroup.  This function is non-blocking, and the
+// internal goroutine will exit when the channel is closed.
+func (r Reconciler) processManagerEvents(events <-chan commonconsumer.ManagerEvent) error {
+	// If the events channel is nil, there's no point listening to it; it will just block forever
+	if events == nil {
+		return fmt.Errorf("no event channel provided")
+	}
+	// Find the namespace and name of the KafkaChannel this dispatcher is monitoring
+	namespace, name, err := cache.SplitMetaNamespaceKey(r.channelKey)
+	if err != nil {
+		return fmt.Errorf("invalid resource key")
+	}
+	key := types.NamespacedName{Namespace: namespace, Name: name}
+	go func() {
+		for event := range events {
+			// Stopping or Starting a consumer group requires a reconciliation in order to adjust the status block
+			// of the KafkaChannel - EnqueueKey will do that
+			switch event.Event {
+			case commonconsumer.GroupStopped:
+				r.logger.Debug("Processing GroupStopped Event From Consumer Group Manager")
+				r.impl.EnqueueKey(key)
+			case commonconsumer.GroupStarted:
+				r.logger.Debug("Processing GroupStarted Event From Consumer Group Manager")
+				r.impl.EnqueueKey(key)
+			}
```

Comment:
Can the log messages include the ConsumerGroup ID ?

Can you add a "default" case that logs unexpected/unsupported event type (just as defensive coding and so that handling of this is clear?)

Reply:
Sure; I added the groupId to a new logger and filled out the rest of the cases